pr_agent/algo/utils.py

```diff
@@ -258,3 +258,26 @@ def update_settings_from_args(args: List[str]) -> List[str]:
             else:
                 other_args.append(arg)
     return other_args
+
+
+def load_yaml(review_text: str) -> dict:
+    review_text = review_text.lstrip('```yaml').rstrip('`')
+    try:
+        data = yaml.load(review_text, Loader=yaml.SafeLoader)
+    except Exception as e:
+        logging.error(f"Failed to parse AI prediction: {e}")
+        data = try_fix_yaml(review_text)
+    return data
+
+def try_fix_yaml(review_text: str) -> dict:
+    review_text_lines = review_text.split('\n')
+    data = {}
+    for i in range(1, len(review_text_lines)):
+        review_text_lines_tmp = '\n'.join(review_text_lines[:-i])
+        try:
+            data = yaml.load(review_text_lines_tmp, Loader=yaml.SafeLoader)
+            logging.info(f"Successfully parsed AI prediction after removing {i} lines")
+            break
+        except:
+            pass
+    return data
```

Comment:
**Suggestion:** In the `try_fix_yaml` function, you are catching a general exception and not doing anything with it. It's a good practice to catch specific exceptions that you expect might be raised in your code. In this case, you should catch `yaml.YAMLError` which is raised by `yaml.load` when it fails to parse a YAML document. Also, it would be beneficial to log the error message for debugging purposes.
```suggestion
def try_fix_yaml(review_text: str) -> dict:
    review_text_lines = review_text.split('
')
    data = {}
    for i in range(1, len(review_text_lines)):
        review_text_lines_tmp = '\n'.join(review_text_lines[:-i])
        try:
            data = yaml.load(review_text_lines_tmp, Loader=yaml.SafeLoader)
            logging.info(f"Successfully parsed AI prediction after removing {i} lines")
            break
        except yaml.YAMLError as e:
            logging.error(f"Failed to parse AI prediction after removing {i} lines: {e}")
    return data
```